tribe-ext-relabeler.php

```diff
@@ -0,0 +1,322 @@
+	 *
+	 * Settings_Helper will append a trailing underscore before each option.
+	 *
+	 * @see \Tribe\Extensions\Example\Settings::set_options_prefix()
+	 *
+	 * @return string
+	 */
+	private function get_options_prefix() {
+		return (string) str_replace( '-', '_', PLUGIN_TEXT_DOMAIN );
+	}
+
+	/**
+	 * Get Settings instance.
+	 *
+	 * @return Settings
+	 */
+	private function get_settings() {
+		if ( empty( $this->settings ) ) {
+			$this->settings = new Settings( $this->get_options_prefix() );
+		}
+
+		return $this->settings;
+	}
+
+	/**
+	 * Extension initialization and hooks.
+	 */
+	public function init() {
+
+		load_plugin_textdomain( 'PLUGIN_TEXT_DOMAIN', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+		if ( ! $this->php_version_check() ) {
+			return;
+		}
+
+		$this->class_loader();
+
+		$this->get_settings();
+
+		// Events.
+		add_filter( 'tribe_event_label_singular', [ $this, 'get_event_single' ] );
+		add_filter( 'tribe_event_label_singular_lowercase', [ $this, 'get_event_single_lowercase' ] );
+		add_filter( 'tribe_event_label_plural', [ $this, 'get_event_plural' ] );
+		add_filter( 'tribe_event_label_plural_lowercase', [ $this, 'get_event_plural_lowercase' ] );
+
+		// Venues.
+		add_filter( 'tribe_venue_label_singular', [ $this, 'get_venue_single' ] );
+		add_filter( 'tribe_venue_label_singular_lowercase', [ $this, 'get_venue_single_lowercase' ] );
+		add_filter( 'tribe_venue_label_plural', [ $this, 'get_venue_plural' ] );
+		add_filter( 'tribe_venue_label_plural_lowercase', [ $this, 'get_venue_plural_lowercase' ] );
+
+		// Organizers.
+		add_filter( 'tribe_organizer_label_singular', [ $this, 'get_organizer_single' ] );
+		add_filter( 'tribe_organizer_label_singular_lowercase', [ $this, 'get_organizer_single_lowercase' ] );
+		add_filter( 'tribe_organizer_label_plural', [ $this, 'get_organizer_plural' ] );
+		add_filter( 'tribe_organizer_label_plural_lowercase', [ $this, 'get_organizer_plural_lowercase' ] );
+	}
+
+	/**
+	 * Check if we have a sufficient version of PHP. Admin notice if we don't and user should see it.
+	 *
+	 * @link https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/ All extensions require PHP 5.6+.
+	 *
+	 * @return bool
+	 */
+	private function php_version_check() {
+		$php_required_version = '5.6';
+
+		if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+			if (
+				is_admin()
+				&& current_user_can( 'activate_plugins' )
+			) {
+				$message = '<p>';
+				$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', PLUGIN_TEXT_DOMAIN ), $this->get_name(), $php_required_version );
+				$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+				$message .= '</p>';
+
+				tribe_notice( PLUGIN_TEXT_DOMAIN . '-php-version', $message, [ 'type' => 'error' ] );
+			}
+
+			return false;
+		}
+
+		return true;
```

Comment:
I think by swapping the logic here a level of indentation can be removed.

```suggestion
		if ( version_compare( PHP_VERSION, $php_required_version, '>=' ) ) {
			return true;
		}
		
		if ( is_admin() && current_user_can( 'activate_plugins' ) ) {
			$message = '<p>';
			$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', PLUGIN_TEXT_DOMAIN ), $this->get_name(), $php_required_version );
			$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
			$message .= '</p>';

			tribe_notice( PLUGIN_TEXT_DOMAIN . '-php-version', $message, [ 'type' => 'error' ] );
		}

		return false;
```

Reply:
@mitogh this might be a good reco for the extension template.